### DISCUSSION TITLE
PGN parsing, comments and variations ignored

### DISCUSSION THREAD

**Discussion by coffeeDev98** [mliebelt/pgn-parser#271] [Q&A]
Hi @mliebelt ,
Wanted to ask if you had made any changes in the pgn-parser package. 

For example: if the PGN is as given below

1. e3 h5 ( 1... f6 ) 2. Bc4 d6 3. Bb5+ Qd7 4. h3 f5 5. e4 g5 6. Qxh5+ { test } Kd8 7. Bc4 b6 8. Be6 a5 9. Bxd7 b5 10. Qe8# *
The comments and variation lines are getting ignored post load.... 

<img width="705" alt="image" src="https://user-images.githubusercontent.com/63295908/188574957-3cbb36e1-f57b-4b83-b18d-42727fe9c6a1.png">
(The image is for illustrative purpose only.)

Any idea on why this can occur? Or if you faced and resolved this issue, can you share the commit# for the same...?

**Comment by mliebelt** [comment]
When you go to https://mliebelt.github.io/PgnViewerJS/examples.html#1110 you will see a game with variations. There was no release for 6 months now, so I cannot think of any change by myself as reason for that behavior.

Can you share any information how you customized the viewer, and which version do you use for that? Do you use the main version?

No idea, what fails, and no possibility to check it for the next 2 weeks

**Comment by mliebelt** [comment]
I added a new test case to the main branch (current state, no changes the last few months), and I showed in debug mode in the IDE that the result is retrieved as expected. See the first screenshot with the move `f6` as variant, and the comment of the 11th move.

![image](https://user-images.githubusercontent.com/520039/191328343-31c0e342-b149-4f06-98f4-e8644be64e86.png)

![image](https://user-images.githubusercontent.com/520039/191328529-10f905fa-0059-42d4-8acd-b83525591de4.png)

Could you elaborate what you did? I am pretty sure that published version of `pgn-parser` is not broken, so which version are you using? What did you do to retrieve the cut of all moves and comments that are not in the main line? Any chance to look at your code then? The pgn-parser currently has no way to strip all but the main line ...

 [Here is the commit](https://github.com/mliebelt/pgn-parser/commit/640a5d416b19b13847e9e462df6638f349877532) that contains the additional test, test assertion is meaningless, but could include the check of the comment / check of the variations.
